File: src/main/java/adapter/repository/ExchangeRateV1Repository.java
package adapter.repository;

import adapter.Currency;
import adapter.ExchangeRateV1;

public class ExchangeRateV1Repository implements ExchangeRateV1 {
    public double fromUSD(Currency to) {
        return switch (to) {
            case KRW -> 1271.14;
            case USD -> 1;
        };
    }

    public double fromKRW(Currency to) {
        return switch (to) {
            case USD -> 0.00079;
            case KRW -> 1;
        };
    }
}

File: kotlin/src/main/kotlin/adapter/repository/ExchangeRateV1Repository.kt
package adapter.repository

import adapter.Currency
import adapter.ExchangeRateV1

class ExchangeRateV1Repository : ExchangeRateV1 {
    override fun fromUSD(to: Currency): Double {
        return when (to) {
            Currency.KRW -> 1271.14
            Currency.USD -> 1.0
        }
    }

    override fun fromKRW(to: Currency): Double {
        return when (to) {
            Currency.USD -> 0.00079
            Currency.KRW -> 1.0
        }
    }
}